eqcorrscan/utils/plotting.py

```diff
@@ -2186,6 +2186,131 @@ def subspace_fc_plot(detector, stachans, **kwargs):
     return fig
 
 
+@additional_docstring(plotting_kwargs=plotting_kwargs)
+def mapplot(events, bgcolor='#909090', method='depth', cpalette='jet_r',
+            s=1, lw=1, marker=',', **kwargs):
+    """
+    Plot seismicity in a 2D map with two cross section along latitude and
+    longitude.
+
+    :type events: list or 'obspy.core.event.catalog.Catalog'
```

Comment:
I'm not keen on supporting two very different things through the same argument. I would just support either a `Catalog` or a list of tuples.

Reply:
When we use some program to locate our events, results of some of them is not a standard catalog and readable with the obspy function, then I think It's good to support this type of inputs.
I can define two different arguments as 'catalog' and 'nodes' to solve this situation and at the same time preserve supporting two type of input.
Is this good?

Reply:
That sounds like a good option: set the default for both `catalog` and `nodes` (maybe call is `locations`?) to `None`, then have a statement in the first line of the code like:
```python
assert (catalog and locations) or catalog or locations, "Requires catalog and/or locations"
locations = locations or []
if catalog:
    for event in catalog:
        try:
            origin = event.preferred_origin() or event.origins[0]
        except IndexError:  # No origin found
            continue
        locations.append(...)
```